src/main/java/com/sucy/skill/dynamic/trigger/ProjectileHitTrigger.java

```diff
@@ -38,10 +39,15 @@ public Class<ProjectileHitEvent> getEvent() {
     @Override
     public boolean shouldTrigger(ProjectileHitEvent event, int level, Settings settings) {
         List<String> projectiles = settings.getStringList("projectile");
-        String type = settings.getString("type","both");
-        boolean hit = !Objects.isNull(event.getHitEntity());
+        String typeBase = settings.getString("type","both");
+        String type = typeBase;
+        if (Stream.of("both","entity","block").noneMatch(c-> typeBase.equalsIgnoreCase(c))){
+            type = "both";
+        }
+        boolean hitEntity = !Objects.isNull(event.getHitEntity());
+        boolean hitBlock = !Objects.isNull(event.getHitBlock());
         boolean isCorrectProjectile = projectiles.isEmpty() || projectiles.contains("Any") || projectiles.stream().anyMatch(projectile->event.getEntityType().name().equalsIgnoreCase(projectile));
-        boolean isCorrectType = type.equalsIgnoreCase("both") || type.equalsIgnoreCase("entity") == hit;
+        boolean isCorrectType = type.equalsIgnoreCase("both") || (type.equalsIgnoreCase("entity") == hitEntity && type.equalsIgnoreCase("block") == hitBlock);
```

Comment:
Shouldn't this be `|| type is entity || type is block` instead of `&&`?

Reply:
`type = "block"`
`hitEntity = false`
`type.equals("entity") == hitEnity` `false == false` and it will return true before reaching the `type is block` check

Reply:
I mean, I guess that works. The readability there is just very difficult.

Reply:
I just realized that this check is also redundant as i am validated the `type` value before so i removed it.